src/store/plugins/load-geojson-style-and-data.plugin.js

```diff
@@ -54,20 +92,33 @@ async function loadDataAndStyle(geoJsonLayer) {
  */
 export default function loadGeojsonStyleAndData(store) {
     store.subscribe((mutation) => {
-        const addLayersSubscriber = async (layers) => {
-            const updatedLayers = await layers
-                .filter((layer) => layer instanceof GeoAdminGeoJsonLayer && layer.isLoading)
-                .map((layer) => {
-                    log.debug(`Loading data/style for added GeoJSON layer`, layer)
-                    return loadDataAndStyle(store, layer)
+        const addLayersSubscriber = (layers) => {
+            layers
+                .filter((layer) => layer instanceof GeoAdminGeoJsonLayer)
+                .forEach((layer) => {
+                    if (layer.isLoading) {
+                        log.debug(`Loading data/style for added GeoJSON layer`, layer)
```

Comment:
Based on other comment, here you could simply keep the `.map()` and do the autoreload in it.